app/utils.py

```diff
@@ -870,6 +898,8 @@ def __init__(self, request, response, env):
         if not model.Repo.get_by_key_name(self.repo):
             if legacy_redirect.do_redirect(self):
                 return legacy_redirect.redirect(self)
+            if self.maybe_redirect_for_repo_alias(request):
```

Comment:
maybe we want to do this before the authorization check, which would be wasted if we decide to redirect?